docs/user/ppl/cmd/lookup.rst

```diff
@@ -0,0 +1,153 @@
+=============
+lookup
+=============
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Description
+============
+| Use the ``lookup`` command to do a lookup from another index and add the fields and values from the lookup document to the search result.
+
+Syntax
+============
+lookup <lookup-index> <lookup-field> [AS <local-lookup-field>] ["," <lookup-field> [AS <local-lookup-field>]]... [appendonly=<bool>] [<source-field> [AS <local-source-field>]] ["," <source-field> [AS <local-source-field>]]...
+
+* lookup-index: mandatory. the name of the lookup index. If more than one is provided, all of them must match.
+* lookup-field: mandatory. the name of the lookup field. Must be existing in the lookup-index. It is used to match to a local field (in the current search) to get the lookup document. When there is no lookup document matching it is a no-op. If there is more than one an exception is thrown.
+* local-lookup-field: optional. the name of a field in the current search to match against the lookup-field. **Default:** value of lookup-field.
+* appendonly: optional. indicates if the values to copy over to the search result from the lookup document should overwrite existing values. If true no existing values are overwritten. **Default:** false.
```

Comment:
why named it appendonly? does overwrite more straitforward?

Reply:
Maybe - I have no personal preference here